src/components/EditModal/TagsInput.tsx

```diff
@@ -0,0 +1,76 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/core"
+import * as React from "react"
+
+import { WordProperties } from "../../@core/api/model"
+
+import Input from "../../common/Input"
+import { StyledTag } from "../../common/Tag"
+
+type Props = {
+  word: WordProperties
+  disabled: boolean
+  onChange: (tags: string[]) => void
+}
+
+export const canAddTag = (currentTag: string, createdTags: string[]) => {
+  if (currentTag.length === 0) return false
+
+  if (createdTags.includes(currentTag)) return false
+
+  return true
+}
+
+const TagsInput: React.FunctionComponent<Props> = ({
+  word,
+  onChange,
+  disabled: isLoading
+}) => {
+  const [currentTag, setCurrentTag] = React.useState<string>("")
+  const [createdTags, setCreatedTags] = React.useState<string[]>(word.tags)
```

Comment:
Do we need to keep this on state? Can we update the word and make apollo immediately refresh it on the page? It might not be that easy with the current setup though... Would do you think?

Reply:
Doesn't look like this is easy (or doable at all). This component is used for creating new words too, so apollo might not even have it in the cache at all.